doc/crypto/BIO_read.pod

```diff
@@ -31,7 +42,10 @@ BIO_puts() attempts to write a NUL-terminated string B<buf> to BIO B<b>.
 
 =head1 RETURN VALUES
 
-All these functions return either the amount of data successfully read or
+BIO_read_ex() and BIO_write_ex() return 1 if data was successfully read or
+written, and 0 otherwise.
```

Comment:
I would prefer it that we just returned 0 on failure and 1 on success...and that is what SSL_read_ex() and SSL_write_ex() do. Perhaps I should change it. Mostly applications don't care what the value of the return is...a failure is a failure...if you want more info look at the error queue.